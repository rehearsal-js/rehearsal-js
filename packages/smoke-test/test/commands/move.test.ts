import { afterEach, describe, expect, test } from 'vitest';
import { cleanOutput, prepareProject, rehearsalCLI } from '../test-helpers/index.js';

import type { Project } from 'fixturify-project';

describe('move command', () => {
  let project: Project;

  afterEach(() => {
    project.dispose();
  });

  test('base_js_app', async () => {
    project = prepareProject('base_js_app');
    await project.write();

    const result = await rehearsalCLI(
      'move',
      '.',
      ['--rootPath', project.baseDir, '--ignore', 'vitest.config.ts,docs/**'],
      {
        cwd: project.baseDir,
      }
    );
    expect(cleanOutput(result.stdout, project.baseDir)).toMatchSnapshot();
  });

  test('ember_js_app_4.11', async () => {
    project = prepareProject('ember_js_app');
    await project.write();

    const result = await rehearsalCLI('move', 'app', ['--rootPath', project.baseDir], {
      cwd: project.baseDir,
    });
    expect(cleanOutput(result.stdout, project.baseDir)).toMatchSnapshot();
  });

  test('glimmerx_js_app', async () => {
    project = prepareProject('glimmerx_js_app');
    await project.write();

    const result = await rehearsalCLI('move', 'src', ['--rootPath', project.baseDir], {
      cwd: project.baseDir,
    });
    expect(cleanOutput(result.stdout, project.baseDir)).toMatchSnapshot();
  });
});
