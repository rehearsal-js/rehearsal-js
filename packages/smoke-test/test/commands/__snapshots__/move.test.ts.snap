// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`move command > base_js_app 1`] = `
"info:    @rehearsal/move<test-version>
[STARTED] Analyzing project dependency graph
[SUCCESS] Analyzing project dependency graph
[STARTED] Executing git mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] renamed: 
[DATA] ./index.js -> ./index.ts
[DATA] ./src/index.js -> ./src/index.ts
[DATA] ./src/foo/baz.js -> ./src/foo/baz.ts
[DATA] ./src/foo/biz.js -> ./src/foo/biz.ts
[DATA] ./src/foo/e.gjs -> ./src/foo/e.gts
[DATA] ./src/foo/buz/biz.js -> ./src/foo/buz/biz.ts
[DATA] ./module-a/index.js -> ./module-a/index.ts
[DATA] ./module-a/src/baz.js -> ./module-a/src/baz.ts
[DATA] ./module-a/src/foo.js -> ./module-a/src/foo.ts
[DATA] ./module-b/index.js -> ./module-b/index.ts
[DATA] ./module-b/src/tires.js -> ./module-b/src/tires.ts
[DATA] ./module-b/src/car.js -> ./module-b/src/car.ts
[SUCCESS] Executing git mv"
`;

exports[`move command > ember_js_app_4.11 1`] = `
"info:    @rehearsal/move<test-version>
[STARTED] Analyzing project dependency graph
[SUCCESS] Analyzing project dependency graph
[STARTED] Executing git mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] renamed: 
[DATA] ./app/app.js -> ./app/app.ts
[DATA] ./app/router.js -> ./app/router.ts
[DATA] ./app/adapters/application.js -> ./app/adapters/application.ts
[DATA] ./app/components/hello-world.gjs -> ./app/components/hello-world.gts
[DATA] ./app/components/map.js -> ./app/components/map.ts
[DATA] ./app/components/rentals.js -> ./app/components/rentals.ts
[DATA] ./app/components/share-button.js -> ./app/components/share-button.ts
[DATA] ./app/models/rental.js -> ./app/models/rental.ts
[DATA] ./app/routes/index.js -> ./app/routes/index.ts
[DATA] ./app/routes/rental.js -> ./app/routes/rental.ts
[DATA] ./app/serializers/application.js -> ./app/serializers/application.ts
[DATA] ./app/services/locale.js -> ./app/services/locale.ts
[DATA] ./app/components/rental/image.js -> ./app/components/rental/image.ts
[DATA] ./app/components/rentals/filter.js -> ./app/components/rentals/filter.ts
[SUCCESS] Executing git mv"
`;

exports[`move command > glimmerx_js_app 1`] = `
"info:    @rehearsal/move<test-version>
[STARTED] Analyzing project dependency graph
[SUCCESS] Analyzing project dependency graph
[STARTED] Executing git mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] renamed: 
[DATA] ./src/GreetingHeader.js -> ./src/GreetingHeader.ts
[DATA] ./src/App.js -> ./src/App.ts
[DATA] ./src/SimpleComponent.js -> ./src/SimpleComponent.ts
[DATA] ./src/index.js -> ./src/index.ts
[SUCCESS] Executing git mv"
`;
